nemo/collections/nlp/models/ner_model.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, Optional
+
+import pytorch_lightning as pl
+import torch
+import torch.nn as nn
+from torch.utils.data import DataLoader
+
+# TODO replace with nemo module
+# from transformers import BertModel
+from nemo.collections.nlp.modules.common.huggingface import BertEncoder
+
+from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
+from nemo.collections.nlp.data.token_classification_dataset import BertTokenClassificationDataset
+
+__all__ = ['NERModel']
+
+from nemo.collections.nlp.modules.common import TokenClassifier
+
+
+
+
+class NERModel(pl.LightningModule):
+    def __init__(
+        self,
+        num_classes,
+        pretrained_model_name='bert-base-cased',
+        activation='relu',
+        log_softmax=True,
+        dropout=0.0,
+        use_transformer_pretrained=True,
+    ):
+        # init superclass
+        super().__init__()
+        self.bert_model = BertEncoder.from_pretrained(pretrained_model_name)
+        self.hidden_size = self.bert_model.config.hidden_size
+        self.tokenizer = NemoBertTokenizer(pretrained_model=pretrained_model_name)
+        self.classifier = TokenClassifier(
+            hidden_size=self.hidden_size,
+            num_classes=num_classes,
+            activation=activation,
+            log_softmax=log_softmax,
+            dropout=dropout,
+            use_transformer_pretrained=use_transformer_pretrained,
+        )
+
+        self.loss = nn.CrossEntropyLoss()
```

Comment:
we'll need to change the loss a little bit to be able to add typing to it - I'll update dev guide